ui/plugin-system/src/model/graph-queries.ts

```diff
@@ -12,26 +12,31 @@
 // limitations under the License.
 
 import { AbsoluteTimeRange, GraphQueryDefinition, UnixTimeMs } from '@perses-dev/core';
+import { useQuery } from 'react-query';
 import { usePlugin } from '../components/PluginLoadingBoundary';
+import {
+  LegacyDatasources,
+  useLegacyDatasources,
+  useTemplateVariableValues,
+  useTimeRange,
+  VariableStateMap,
+} from '../runtime';
 
 /**
  * A plugin for running graph queries.
  */
 export interface GraphQueryPlugin<Spec = unknown> {
-  useGraphQuery: UseGraphQueryHook<Spec>;
+  getGraphData: (definition: GraphQueryDefinition<Spec>, ctx: GraphQueryContext) => Promise<GraphData>;
```

Comment:
If everyone is cool with it, I'm gonna do it in a follow-up PR just since it'll be easier to track in the changelog as another breaking change. Totally agree the name kinda sucks. 😂